packages/builder/src/helpers/data/datasourceSelectFormat.js

```diff
@@ -0,0 +1,19 @@
+const table = (table, datasources) => ({
```

Comment:
These formatters are to ensure that tables/views selected for generated templates _exactly_ match the expected format of the settings control/client prop they get fed into. 

Feeding in the raw table or view actually works, but the table selector will for instance trigger an update to massage the table into the format it expects, and when it does this it can trigger unexpected updates to other settings that depend on the table.